pkg/test/env/istio.go

```diff
@@ -87,7 +88,8 @@ var (
 	IstioRoot = path.Join(GOPATH.ValueOrDefault(build.Default.GOPATH), "/src/istio.io/istio")
 
 	// ChartsDir is the Kubernetes Helm chart directory in the repository
-	ChartsDir = path.Join(IstioRoot, "install/kubernetes/helm")
+//	ChartsDir = path.Join(IstioRoot, "install/kubernetes/helm")
+	ChartsDir = path.Join("/work/install/kubernetes/helm")
```

Comment:
Does 

```
func getDefaultIstioTop() string {
	// Assume it is run inside istio.io/istio
	current, err := os.Getwd()
	if err != nil {
		panic(err)
	}
	idx := strings.Index(current, filepath.Join("/src", "istio.io", "istio"))
	if idx > 0 {
		return current[0:idx]
	}
	return current // launching from GOTOP (for example in goland)
}
```

can we not make this function return the correct thing with/without build container?

Reply:
good suggestion @howardjohn. I have consolidated the 3 variables `IstioSrc`, `IstioRoot`, and `IstioTop` into `IstioTop` and modified this function slightly to detect if we are running in a container. If we are, `/work` is used as the src dir.